api/src/views/context.rs

```diff
@@ -79,13 +91,57 @@ impl UserContext {
                 user: None,
             }),
             // User is logged in and initialized, return the full user data
-            Some((Some(user_id), Some(username))) => Some(Self {
-                user_provider_id,
-                user: Some(AuthorizedUser {
-                    id: user_id,
-                    username,
-                }),
-            }),
+            Some((Some(user_id), Some(username))) => {
+                // Fetch all of the user's roles and permissions
+                let permissions_query_result = users::table
+                    .find(user_id)
+                    // yo dawg i heard you like joins
+                    .inner_join(
+                        user_roles::table.left_join(
+                            // LEFT JOIN here so that even if the role has
+                            // no
+                            // permissions, we still get its is_admin value
+                            roles::table.left_join(
+                                role_permissions::table
+                                    .left_join(permissions::table),
+                            ),
+                        ),
+                    )
+                    .select((
+                        permissions::columns::name.nullable(),
+                        roles::columns::is_admin,
+                    ))
+                    .distinct()
+                    // .filter(users::columns::id.eq(user_id))
+                    .get_results::<(Option<String>, bool)>(conn)?;
+
+                // For permissions, just collect em all
+                // For is_admin, do an or-map (true if any are true)
+                let (permissions, is_admin) = permissions_query_result
+                    .into_iter()
+                    .try_fold::<_, _, ResponseResult<_>>(
+                    (HashSet::new(), false),
+                    |(mut permissions, is_admin),
+                     (permission_opt, is_admin_row)| {
+                        if let Some(permission) = permission_opt {
+                            // Parsing here shouldn't ever fail because
+                            // PermissionType covers all DB values
+                            permissions.insert(permission.parse()?);
+                        }
+                        Ok((permissions, is_admin || is_admin_row))
+                    },
+                )?;
+
+                Some(Self {
+                    user_provider_id,
+                    user: Some(AuthorizedUser {
+                        id: user_id,
+                        username,
+                        permissions,
+                        is_admin,
+                    }),
+                })
+            }
```

Comment:
@JRMurr On the last pr i know you asked if we can combine this. It's definitely possible in SQL but right now diesel doesn't support the aggregations we need properly. I'm pretty sure it will work in diesel 2.0 so I left a note to fix that once 2.0 gets release and we upgrade.